src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -176,23 +176,15 @@ private void RequestRefresh(AuthenticationTicket ticket)
 
         private CookieOptions BuildCookieOptions()
         {
-            var cookieOptions = new CookieOptions
-            {
-                Domain = Options.CookieDomain,
-                SameSite = Options.CookieSameSite,
-                HttpOnly = Options.CookieHttpOnly,
-                Path = Options.CookiePath ?? (OriginalPathBase.HasValue ? OriginalPathBase.ToString() : "/"),
-            };
+            var cookieOptions = Options.Cookie.Build(Context);
+            // ignore the 'Expires' value as this will be computed elsewhere
```

Comment:
The behavior here is funky, and FWIW one of the reasons I was suggesting we leave `Expiration` off CookeBuilder. This handler takes into consideration `CookieSignInContext.Properties.ExpiresUtc` and `CookieSignInContext.Properties.IsPersistent` to calculate `CookieOptions.Expires`. So, the value of `.Expires` set by the builder needs to be unset to avoid changing behavior.

https://github.com/aspnet/Security/blob/namc/cookie-options/test/Microsoft.AspNetCore.Authentication.Test/CookieTests.cs#L141

It would be good to find ways to clean this up without breaking existing behavior.